core/src/main/java/com/linecorp/armeria/common/ContentTooLargeException.java

```diff
@@ -23,20 +23,63 @@
 
     private static final long serialVersionUID = 4901614315474105954L;
 
+    private long transferred;
+    private long total;
+    private long maximum;
+
     private static final ContentTooLargeException INSTANCE = new ContentTooLargeException(false);
 
     /**
      * Returns a {@link ContentTooLargeException} which may be a singleton or a new instance, depending on
      * {@link Flags#verboseExceptionSampler()}'s decision.
+     * @deprecated use {@code builder} instead.
      */
+    @Deprecated
     public static ContentTooLargeException get() {
         return Flags.verboseExceptionSampler().isSampled(ContentTooLargeException.class) ?
                new ContentTooLargeException() : INSTANCE;
     }
 
+    /**
+     * Returns a {@link ContentTooLargeExceptionBuilder} which may return a singleton or a new instance in its
+     * {@code build} method, depending on {@link Flags#verboseExceptionSampler()}'s decision.
+     */
+    public static ContentTooLargeExceptionBuilder builder() {
+        return new ContentTooLargeExceptionBuilder();
+    }
+
     private ContentTooLargeException() {}
 
     private ContentTooLargeException(@SuppressWarnings("unused") boolean dummy) {
         super(null, null, false, false);
     }
+
+    ContentTooLargeException(long transferred, long total, long maximum) {
+        super(String.format("content length too large: %d + %d > %d", transferred, total, maximum));
```

Comment:
Is it human-readable to leave the numbers with their name such as:

```suggestion
        super(String.format("content length too large: transferred(%d) + total(%d) > maximum(%d)", transferred, total, maximum));
```